root/frontend/src/components/Weather/Weather.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import useStyles from './Weather.style';
+
+function Weather() {
+  const classes = useStyles();
+  const fetchWeatherData = async (): Promise<any> => {
+    const data = await axios('http://localhost:5005/weather/getWeatherInfo');
+    return data.data;
+  };
+
+  type WeatherDataObj = {
+    current: any;
+    location: any;
+  };
+
+  const [weatherData, setWeatherData] = useState<WeatherDataObj>();
+
+  useEffect(() => {
+    const getWeather = async () => {
+      const fetchedWeatherData = await fetchWeatherData();
+      setWeatherData(fetchedWeatherData);
+    };
+    getWeather();
```

Comment:
While this is how I would also handle running an async operation inside a `useEffect`, there is something to note about doing this. Since the operation is run without an await, we can never predict when this function will finish relative to what's around it. I believe we need at least an unmount check to ensure that whatever promise was made here is settled before we unmount, thus preventing a memory leak. I've seen quite a few ways to go about this, so let me know if you need any help with implementing this.